src/codegen/mod.rs

```diff
@@ -3907,9 +3923,14 @@ mod utils {
             //
             // [1]: http://c0x.coding-guidelines.com/6.7.5.3.html
             let arg_ty = match *arg_ty.canonical_type(ctx).kind() {
-                TypeKind::Array(t, _) => {
-                    t.to_rust_ty_or_opaque(ctx, &())
-                        .to_ptr(ctx.resolve_type(t).is_const())
+                TypeKind::Array(t, size) => {
+                    let stream = t.to_rust_ty_or_opaque(ctx, &());
```

Comment:
Is there any reason this can just use `arg_item.to_rust_ty_or_opaque().to_ptr()` if the new option is true instead of adding a new trait?

Reply:
hmmmm but how will it get the size of the array? or is the size also part of the item and not just in the enum?

Reply:
arg_item is the array, right? So it should just work, as far as I can tell, the `kind()` is part of the item.

Reply:
Yep. took me some time to figure out the different `to_rust_ty` trait for this and what it expects.